components/map/basic/src/index.js

```diff
@@ -0,0 +1,257 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import LeafletMap from './leaflet/map'
+import { mapViewModes, NO_OP } from './leaflet/constants'
+
+class MapBasic extends Component {
+  constructor (props) {
+    super(props)
+    this.setMapEventDefinition(props)
+    this.mapInstance = undefined
+    this.isHeatmapVisible = false
+    this.isSatelliteView = false
+  }
+
+  setMapEventDefinition () {
+    this.mapEventList = [
+      {
+        name: 'leaflet_map_click',
+        handleFunction: (evt) => this.props.onMapClick(evt.detail)
+      },
+      {
+        name: 'leaflet_map_dragend',
+        handleFunction: (evt) => this.props.onMapDragEnd(evt.detail)
+      },
+      {
+        name: 'leaflet_map_loaded',
+        handleFunction: (evt) => this.props.onMapLoad(evt.detail)
+      },
+      {
+        name: 'leaflet_map_zoomend',
+        handleFunction: (evt) => this.props.onMapZoomEnd(evt.detail)
+      },
+      {
+        name: 'leaflet_map_layer_normal',
+        handleFunction: (evt) => this.props.onNormalView(evt.detail)
+      },
+      {
+        name: 'leaflet_map_poiclick',
+        handleFunction: (evt) => this.props.onPoiClick(evt.detail)
+      },
+      {
+        name: 'leaflet_map_poimouseout',
+        handleFunction: () => this.props.onPoiMouseOut()
+      },
+      {
+        name: 'leaflet_map_poimouseover',
+        handleFunction: (evt) => this.props.onPoiMouseOver(evt.detail)
+      },
+      {
+        name: 'leaflet_map_layer_satellite',
+        handleFunction: (evt) => this.props.onSatelliteView(evt.detail)
+      }
+    ]
+  }
+
+  getMapConfig () {
+    return {
+      id: this.props.id,
+      heatMapUrl: this.props.heatMapUrl,
+      latitude: this.props.center[0],
+      longitude: this.props.center[1],
+      literals: this.props.literals,
+      maxZoom: this.props.maxZoom,
+      minZoom: this.props.minZoom,
+      polygons: this.props.polygons,
+      showHeatmap: this.props.showHeatmap,
+      showSatelliteView: this.props.showSatelliteView,
+      mapViewModes: this.props.mapViewModes,
+      selectedMapViewMode: this.props.selectedMapViewMode,
+      zoomControl: this.props.zoomable,
+      zoomControlPosition: this.props.zoomControlPosition,
+      icons: this.props.icons,
+      enableViewMenu: this.props.enableViewMenu,
+      dragging: this.props.isInteractable,
+      zoom: this.props.zoom,
+      appId: this.props.appId,
+      appCode: this.props.appCode
+    }
+  }
+
+  subscribeToMapEvents () {
+    this.mapEventList.forEach(mapEvent => this.mapDOMInstance.addEventListener(mapEvent.name, mapEvent.handleFunction))
+  }
+
+  unsubscribeFromMapEvents () {
+    this.mapEventList.forEach(mapEvent => this.mapDOMInstance.removeEventListener(mapEvent.name, mapEvent.handleFunction))
+  }
+
+  checkWhichViewShouldBeDisplayed (showSatelliteView) {
```

Comment:
You're not checking, you're showing the correct view... The name is not good to me.

Reply:
This is not true at all. This fn will check if there's a view to be loaded. If there is will load it but if for example we call again this FN and nothing changes the function won't do anything.